mesh/nurbs.cpp

```diff
@@ -4804,6 +4901,41 @@ const Array<int>& NURBSExtension::GetPatchBdrElements(int patch)
    return patch_to_bel[patch];
 }
 
+NURBSPatch::NURBSPatch(const KnotVector *kv0, const KnotVector *kv1, int dim_,
+                       const real_t* control_points)
+{
+   kv.SetSize(2);
+   kv[0] = new KnotVector(*kv0);
+   kv[1] = new KnotVector(*kv1);
+   init(dim_);
+   memcpy(data, control_points, sizeof (real_t) * ni * nj * dim_);
```

Comment:
`MFEM_ASSERT(ni > 0 && nj > 0 && dim_ > 0, "Invalid dimensions")`

Reply:
I put equivalent checks in the `init` method instead, up to one small difference that the control point dimension must be greater than 1 instead of 0, because it includes the weight component. (I don't think it will break any memory allocation/copying at this stage to have a weight-only spline, but I'm not sure what the interpretation is.)